src/Domain/Composer/Version/DrupalCoreVersionResolver.php

```diff
@@ -400,4 +403,20 @@ private function convertToDev(string $version): string {
     return implode('.', $parts);
   }
 
+  /**
+   * Validates if the version is "NEXT MAJOR LATEST MINOR BETA OR LATER".
+   *
+   * @param string $version
+   *   The version to validate.
+   *
+   * @throws \Acquia\Orca\Exception\OrcaVersionNotFoundException
+   */
+  private function validateNextMajorLatestMinorBetaOrLater($version): void {
+    $parts = explode('.', $version);
+    $stability = end($parts);
+    if ((strpos($stability, "alpha") !== FALSE) || (strpos($stability, "dev") !== FALSE)) {
```

Comment:
Let's have some more [good, expressive variable names](https://wiki.c2.com/?GoodVariableNames=) to make the logic clearer here. For example:

```php
$is_alpha = strpos($stability, "alpha") !== FALSE;
```